lib/wsdl.js

```diff
@@ -1528,31 +1528,30 @@ WSDL.prototype.objectToDocumentXML = function(name, params, nsPrefix, nsURI, typ
  * @param {String} nsURI
  * @returns {string}
  */
-WSDL.prototype.objectToRpcXML = function(name, params, nsPrefix, nsURI) {
-  var parts = [];
-  var defs = this.definitions;
-  var nsAttrName = '_xmlns';
-
-  nsPrefix = nsPrefix || findPrefix(defs.xmlns, nsURI);
-  nsURI = nsURI || defs.xmlns[nsPrefix];
-  nsPrefix = nsPrefix === TNS_PREFIX ? '' : (nsPrefix + ':');
-  parts.push(['<', nsPrefix, name, '>'].join(''));
-
-  for (var key in params) {
-    if (!params.hasOwnProperty(key)) {
-      continue;
-    }
-    if (key !== nsAttrName) {
-      var value = params[key];
-      parts.push(['<', key, '>'].join(''));
-      parts.push((typeof value === 'object') ? this.objectToXML(value, key, nsPrefix, nsURI) : xmlEscape(value));
-      parts.push(['</', key, '>'].join(''));
-    }
-  }
-  parts.push(['</', nsPrefix, name, '>'].join(''));
-
-  return parts.join('');
-};
+ WSDL.prototype.objectToRpcXML = function(name, params, nsPrefix, nsURI,isParts) {
+   var parts = [];
+   var defs = this.definitions;
+   var nsAttrName = '_xmlns';
+
+   nsPrefix = nsPrefix || findPrefix(defs.xmlns, nsURI);
+
+   nsURI = nsURI || defs.xmlns[nsPrefix];
+   nsPrefix = nsPrefix === TNS_PREFIX ? '' : (nsPrefix + ':');
+
+   parts.push(['<', nsPrefix, name, '>'].join(''));
+
+   for (var key in params) {
+     if (!params.hasOwnProperty(key)) continue;
+     if (key !== nsAttrName) {
+       var value = params[key];
+       parts.push(['<', (isParts ? '' : nsPrefix), key, '>'].join(''));
```

Comment:
Instead of having to check `isParts ? '' : nsPrefix` two times, wouldn't it be also sufficient to create athe key right away like 
```javascript
var value = params[key];
var prefixedKey = (isParts ? '' : nsPrefix) + key;
parts.push(['<', prefixedKey, '>].join('')); 
parts.push((typeof value === 'object') ? this.objectToXML(value, key, nsPrefix, nsURI) : xmlEscape(value));
parts.push(['</', prefixedKey, '>].join(''));
```